dlt/extract/resource.py

```diff
@@ -343,7 +344,9 @@ def _gen_wrap(gen: TPipeStep) -> TPipeStep:
                 for i in gen:  # type: ignore # TODO: help me fix this later
                     yield i
                     if i is not None:
-                        count += 1
+                        with counter_lock:
```

Comment:
@sultaniman I think all this code is run on the main pipeline thread anyway, so the lock will not help here, but I'm not 100% sure

Reply:
Tbh I also didn't dig deeper, my theory was that we submit the work to `ThreadPoolExecutor` which can potentially case data races.

Reply:
operations on integers in CPython are atomic so we are not locking anything. IMO race condition goes deep here and we are better off by making our tests to accept 10th element 🤷